File: Java/12-pattern-number-crown.java
import java.util.Scanner;

public class Solution {
    public static void numberCrown(int n) {
        // Write your code here.
        for(int i = 0; i<n; i++){
            for(int j = 0; j<i+1; j++)
                System.out.print(j+1 + " ");
            
            for(int j = 0; j<n+1; j++)
                System.out.print(" ");

            for(int j = i+1; j>0; j--)
                System.out.print(j + " ");

            System.out.println();
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        numberCrown(n);
        scanner.close();
    }
}

File: Python/12-pattern-number-crown.py
def numberCrown(n):
    # Write your code here.
    for i in range(n):
        for j in range(i + 1):
            print(j + 1, end=" ")

        for j in range(n + 1):
            print(" ", end=" ")

        for j in range(i + 1, 0, -1):
            print(j, end=" ")

        print()

def main():
    n = int(input())
    numberCrown(n)

if __name__ == "__main__":
    main()
